### PR TITLE
[FIX] Selection: Altering selection should scroll the viewport 

### PR DESCRIPTION
**Pull request by rrahir**
## Description:

description of this task, what is implemented and why it is implemented that way.

Odoo task ID : [3050101](https://www.odoo.com/web#id=3050101&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/plugins/ui/autofill.ts [hunk 1]
```diff
@@ -363,6 +368,10 @@ export class AutofillPlugin extends UIPlugin {
     this.autofillZone = { top, bottom, left, right };
   }
 
+  private saveSteps(steps: number) {
```
Comment (LucasLefevre) [inline src/plugins/ui/autofill.ts:371]: do we really need a one line method for this?
What do you think of inlining the line where the method is used?

File: src/plugins/ui/autofill.ts [hunk 2]
```diff
@@ -226,12 +227,12 @@ export class AutofillPlugin extends UIPlugin {
     }
 
     if (apply) {
-      const zone = union(this.getters.getSelectedZone(), this.autofillZone);
       this.autofillZone = undefined;
+      this.selection.resizeAnchorZone(this.direction, this.steps);
```
Comment (LucasLefevre) [inline src/plugins/ui/autofill.ts:231]: yep, looks like what it should have been from the start

File: src/selection_stream/selection_stream_processor.ts [hunk 3]
```diff
@@ -142,7 +142,10 @@ export class SelectionStreamProcessor
   /**
    * Set the selection to one of the cells adjacent to the current anchor cell.
    */
-  moveAnchorCell(direction: SelectionDirection, step: SelectionStep = "one"): DispatchResult {
+  moveAnchorCell(direction: SelectionDirection, step: SelectionStep = 1): DispatchResult {
+    if (step !== "end" && step <= 0) {
```
Comment (LucasLefevre) [inline src/selection_stream/selection_stream_processor.ts:146]: Does it work when step=0 ? Probably yes, but it's worth a test

File: tests/components/autofill.test.ts [hunk 4]
```diff
@@ -280,7 +280,6 @@ describe("Autofill component", () => {
     setSelection(parent.model, ["A1:A100"]);
     parent.model.dispatch("SET_VIEWPORT_OFFSET", { offsetX: 400, offsetY: 400 });
     const firstViewport = parent.model.getters.getActiveSnappedViewport();
-    console.log(firstViewport);
```
Comment (LucasLefevre) [inline tests/components/autofill.test.ts:283]: Oh ben tiens :smile:

File: src/helpers/zones.ts [hunk 5]
```diff
@@ -487,7 +487,7 @@ export function mergeOverlappingZones(zones: Zone[]) {
 export function findCellInNewZone(
   oldZone: Zone,
   currentZone: Zone,
-  viewport: Viewport
+  viewport?: Viewport
```
Comment (LucasLefevre) [inline src/helpers/zones.ts:490]: maybe update the docstring to help understanding what impact the viewport has

**Review by LucasLefevre — APPROVED** [review]
only details

File: src/plugins/ui/autofill.ts (hunk 1 above)
Comment (rrahir) [reply src/plugins/ui/autofill.ts:371]: sure, seems sufficient enough

File: tests/components/autofill.test.ts (hunk 4 above)
Comment (rrahir) [reply tests/components/autofill.test.ts:283]: chut :D

**Review by LucasLefevre — APPROVED** [review]
robodoo r+

**Pull request closed without merging**